src/rdkafka_topic.c

```diff
@@ -1518,6 +1519,12 @@ void rd_kafka_topic_scan_all(rd_kafka_t *rk, rd_ts_t now) {
 
                 rd_kafka_topic_wrlock(rkt);
 
+                if (ignore_notexists &&
+                    rkt->rkt_state == RD_KAFKA_TOPIC_S_NOTEXISTS) {
+                        rd_kafka_topic_wrunlock(rkt);
+                        continue;
+                }
```

Comment:
We may want to check app_refcnt here. If a deleted topic gets recreated, or if the application creates a new topic handle, it may make sense to keep fetching state. Though ideally, not every second.